subprojects/core/src/main/java/org/gradle/initialization/DefaultLegacyTypesSupport.java

```diff
@@ -16,25 +16,53 @@
 
 package org.gradle.initialization;
 
+import kotlin.Metadata;
+import kotlin.jvm.JvmClassMappingKt;
+import kotlin.jvm.JvmName;
+import kotlin.jvm.JvmStatic;
+import kotlin.reflect.KClass;
 import org.gradle.api.GradleException;
+import org.gradle.api.plugins.ExtensionContainer;
+import org.gradle.api.plugins.internal.DelegatingConventionInterface;
 import org.gradle.internal.classloader.ClassLoaderUtils;
+import org.jspecify.annotations.Nullable;
+import org.objectweb.asm.AnnotationVisitor;
 import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.MethodVisitor;
 import org.objectweb.asm.Opcodes;
 import org.objectweb.asm.Type;
 
 import java.io.BufferedReader;
 import java.io.IOException;
 import java.io.InputStreamReader;
 import java.net.URL;
-import java.util.HashSet;
+import java.util.LinkedHashSet;
 import java.util.Set;
 
+import static org.objectweb.asm.Type.getMethodDescriptor;
+import static org.objectweb.asm.Type.getType;
+
 /**
  * Enriches class loading with empty interfaces for certain types that have been removed,
  * but which are baked into the bytecode generated by the Groovy compiler.
  */
 public class DefaultLegacyTypesSupport implements LegacyTypesSupport {
-    private static final Type OBJECT_TYPE = Type.getType(Object.class);
+    public static final Type OBJECT_TYPE = getType(Object.class);
```

Comment:
Is this a good place for this code generation?

Reply:
I think so. We are indeed supporting legacy types, but this time in a more complex way. Maybe we should explain the intent here, (or link the comments from DelegatingConventionInterface).